### PR TITLE
Added A Corner Kick Play

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
- added `CornerKickPlay`
- renamed the `AI::Passing` namespace to `Passing` to prevent naming conflicts with the `AI` class

### Testing Done
- grsim testing
- field testing
- currently *NO* unit testing, this will need to get done (though not sure exactly how) before merging 

### Length Justification
Renamed a lot of stuff.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by henryymliu** [comment]
Should I hold off on review until most of the TODOs are resolved (or are they going to be part of different PRs?) and unit tests are written, along with more field testing?

File: src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+{
+    // TODO: Rough description of what this function does (stages, tactics, etc.) here
+
+    // Figure out if we're taking the kick from the +y or -y corner
+    bool kick_from_pos_corner = world.ball().position().y() > 0;
+
+    // We want the two cherry pickers to be in rectangles on the +y and -y sides of the
+    // field in the +x half. We also further offset the rectangle from the goal line
+    // for the cherry-picker closer to where we're taking the corner kick from
+    Vector enemy_defense_area_pos_y_x_offset(world.field().enemyDefenseArea().width(), 0);
```
Comment (MathewMacDougall) [inline src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp]: Maybe `pos_y_goalline_offset` is a better name here? This offset isn't really related to the defense area other than that's related to its value. It's an offset from the goalline, so I think this name is a bit more descriptive.

**Review by MathewMacDougall — COMMENTED** [review]
Did a quick look over this and it looks solid. Ping me once the TODOs are resolved and you want re-review.

**Comment by garethellis0** [comment]
@henrymliu - yeah, I think just hold off on review until I get this cleaned up (currently targeting wednesday evening)

File: src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp (hunk 1 above)
Comment (garethellis0) [reply src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp]: changed to `pos_y_goalline_x_offset`

**Comment by garethellis0** [comment]
@henrymliu @akhilveeraghanta @MathewMacDougall - ready for review!

**Review by akhilveeraghanta — COMMENTED** [review]
The one time pass + score thing is pretty awesome, and its great that we try to bamboozle the other team with the bait, just noticed some minor stuff when trying it out in grsim

cornerkick

Seems like the robot just bumps into the ball, maybe a pivot might be more applicable in this case over a move as it would rotate around the ball? (although this may be fixed once the navigator avoids the ball as an obstacle)

Also noticed that the robot doesn't orient itself properly to the ball, but this could also be a grsim motion controller issue.
 And im not sure if this is really an issue, but the play never seems to finish, stuck in some sort of loop, I can switch from example play to friendly kickoff and back, but I can't switch to corner kick and then switch back to example play (through param window + override ai play set to true), even though I get the debug "finished" msg.

File: src/thunderbots/software/ai/world/game_state.h [hunk 2]
```diff
@@ -229,7 +229,7 @@ class GameState
      *
      * @return true if opposing side is tkaing a direct free kick.
```
Comment (henryymliu) [inline src/thunderbots/software/ai/world/game_state.h]: uber nit: taking?

**Review by henryymliu — COMMENTED** [review]
Nothing to add so far.

**Comment by MathewMacDougall** [comment]
@akhilveeraghanta 	those are all great points.

The robot bumping into the ball is because the navigator isn't merged yet. We have tested with the navigator and confirmed that the robot does not bump the ball, but moves nicely behind it (yay Jon).

The orientation is also a grsim issue. The controllers we wrote for grsim are not great (feel like the got worse tbh). Again, we confirmed this works as intended on the field.

I'm not sure what's up with the Play switching, but if it's not switching when you manually override that likely some other issue (or hopefully a one-time glitch?). Not something to worry about for now.

File: src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    // is taking place to pull enemies away from the goal
+    Point opposite_corner_to_kick = kick_from_pos_corner ? world.field().enemyCornerNeg()
+                                                         : world.field().enemyCornerPos();
+    Point bait_move_tactic_1_pos =
+        opposite_corner_to_kick - Vector(world.field().enemyDefenseArea().width() * 0.5,
+                                         copysign(0.5, opposite_corner_to_kick.y()));
+    Point bait_move_tactic_2_pos =
+        opposite_corner_to_kick - Vector(world.field().enemyDefenseArea().width() * 1.5,
+                                         copysign(0.5, opposite_corner_to_kick.y()));
+    auto bait_move_tactic_1 = std::make_shared<MoveTactic>(true);
```
Comment (MathewMacDougall) [inline src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp:109]: Were we going to make these patrol instead?

File: src/thunderbots/software/ai/world/game_state.h (hunk 2 above)
Comment (MathewMacDougall) [inline src/thunderbots/software/ai/world/game_state.h]: bump

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Minor comments otherwise LGTM

**Comment by garethellis0** [comment]
@akhilveeraghanta - that play switching issue is funky. The expected behavior there would be that setting a play to run would force that play to run, regardless of what state the current play is in. Could you please file a ticket for that with your repro case to make sure we look at this after comp.?

File: src/thunderbots/software/ai/passing/evaluation.cpp [hunk 4]
```diff
@@ -29,6 +29,7 @@ double AI::Passing::ratePass(const World& world, const AI::Passing::Pass& pass,
     double enemy_pass_rating = ratePassEnemyRisk(world.enemyTeam(), pass);
 
     double shoot_pass_rating = ratePassShootScore(world.field(), world.enemyTeam(), pass);
+//    double shoot_pass_rating = 1;
```
[the remark this answers is not in the document]
Comment (garethellis0) [reply src/thunderbots/software/ai/passing/evaluation.cpp]: Oh yup, removed!

File: src/thunderbots/software/ai/world/game_state.h (hunk 2 above)
Comment (garethellis0) [reply src/thunderbots/software/ai/world/game_state.h]: good catch, changed!

File: src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp (hunk 3 above)
Comment (garethellis0) [reply src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp:109]: We can, not sure what the best patrol strategy would be? Moving over to slack real quick.........

**Review by MathewMacDougall — DISMISSED** [review]
LGTM

**Review by henryymliu — DISMISSED** [review]

**Review by MathewMacDougall — APPROVED** [review]

**Pull request merged**